stackclub/taster.py

```diff
@@ -0,0 +1,162 @@
+        """
+        for datasettype in datasettypes:        
+            try:
+                datasetkeys = self.butler.getKeys(datasettype)
+                onekey = list(datasetkeys.keys())[0]
+                metadata = self.butler.queryMetadata(datasettype, [onekey])
+                if self.vb: print("{} dataset exists.".format(datasettype))
+                self.exists[datasettype] = True
+            except:
+                if self.vb: print("{} dataset doesn't exist.".format(datasettype))
+                self.exists[datasettype] = False
+        return
+    
+    def look_for_skymap(self):
+        """
+        Check for the existence of a skymap. 
+        """
+        try:
+            self.skyMap = self.butler.get('deepCoadd_skyMap')
+            self.exists['deepCoadd_skyMap'] = True
+            if self.vb: print("deepCoadd_skyMap exists.")
+        except:
+            self.skyMap = None
+            self.exists['deepCoadd_skyMap'] = False
+            if self.vb: print("deepCoadd_skyMap doesn't exist.")
+        return
+    
+    def estimate_sky_area(self):
+        """
+        Use available skymap to estimate sky area covered by tracts and patches.
+        
+        Returns
+        =======
+        area: float
+            Sky area in square degrees
+        """
+        if self.skyMap is None: return None
+        
+        area_label = 'Total Sky Area (deg$^2$)'
+        if area_label in self.counts.keys():
+            return self.counts[area_label]
+        
+        # Collect tracts from files
+        import os, glob
+        tracts = sorted([int(os.path.basename(x)) for x in
+                 glob.glob(os.path.join(self.repo, 'deepCoadd-results', 'merged', '*'))])
+        self.counts['Number of Tracts'] = len(tracts)
+        # Note: We'd like to do this with the butler, but it appears 'tracts' have to be
+        #       specified in the dataId to be queried, so the queryMetadata method fails
+
+        # Calculate area from all tracts
+        total_area = 0.0  #deg^2
+        plotting_vertices = []
+        for test_tract in tracts:
+            # Get inner vertices for tract
+            tractInfo = self.skyMap[test_tract]
+            vertices = tractInfo._vertexCoordList
+            plotting_vertices.append(vertices)
+
+            # Calculate area of box
+            av_dec = 0.5 * (vertices[2][1] + vertices[0][1])
+            av_dec = av_dec.asRadians()
+            delta_ra_raw = vertices[0][0] - vertices[1][0] 
+            delta_ra = delta_ra_raw.asDegrees() * np.cos(av_dec)
+            delta_dec= vertices[2][1] - vertices[0][1]
+            area = delta_ra * delta_dec.asDegrees()
+
+            # Combine areas
+            total_area += area
+
+        if self.vb: print(area_label, ": ", total_area)
+
+        # Round of the total area for table purposes
+        self.counts[area_label] = round(total_area, 2)
+        return self.counts[area_label]
+
+    def count_things(self):
```

Comment:
I've added an explicit reference to the calexp in the method docstring. This isn't perfect, but good enough for now.

I can imagine a future function that would allow the user to specify the dataset type that they are querying.

Reply:
#174